src/python/pants/backend/python/tasks2/python_repl.py

```diff
@@ -0,0 +1,57 @@
+# coding=utf-8
+# Copyright 2016 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import tempfile
+
+from pex.pex_info import PexInfo
+
+from pants.backend.python.targets.python_requirement_library import PythonRequirementLibrary
+from pants.backend.python.targets.python_target import PythonTarget
+from pants.backend.python.tasks2.python_execution_task_base import PythonExecutionTaskBase
+from pants.task.repl_task_mixin import ReplTaskMixin
+from pants.util.dirutil import safe_rmtree
+
+
+class PythonRepl(ReplTaskMixin, PythonExecutionTaskBase):
+
+  @classmethod
+  def register_options(cls, register):
+    super(PythonRepl, cls).register_options(register)
+    # TODO: Create a python equivalent of register_jvm_tool, and use that instead of these
+    # ad-hoc options.
+    register('--ipython', type=bool,
+             help='Run an IPython REPL instead of the standard python one.')
+    register('--ipython-entry-point', advanced=True, default='IPython:start_ipython',
+             help='The IPython REPL entry point.')
+    register('--ipython-requirements', advanced=True, type=list, default=['ipython==1.0.0'],
+             help='The IPython interpreter version to use.')
+
+  @classmethod
+  def select_targets(cls, target):
+    return isinstance(target, (PythonTarget, PythonRequirementLibrary))
+
+  def extra_requirements(self):
+    if self.get_options().ipython:
+      return [self.get_options().ipython_requirements]
+    else:
+      return []
+
+  def setup_repl_session(self, targets):
+    tmpdir = tempfile.mkdtemp()
```

Comment:
Yes, good idea. I completely neglected to do invalidation etc. I think I figured that this is always fast, but that's not true if resolving extra requirements, so I've done this properly now.

While doing so realized I can get rid of the custom fingerprint strategy for python requirements that I had - the normal strategy will do the right thing via the PythonRequirementLibrary's payload.